superset-frontend/src/components/Select/Select.tsx

```diff
@@ -44,7 +43,21 @@ import { hasOption } from './utils';
 
 const { Option } = AntdSelect;
 
-type AntdSelectAllProps = AntdSelectProps<AntdSelectValue>;
+export type RawValue = string | number;
+
+export interface LabeledValue {
+  key?: string;
+  value: RawValue;
+  label: ReactNode;
```

Comment:
The idea when introducing the `customLabel` prop was that `label` would always be a `string` that could be searched/compared and `customLabel` would be used when you want a `React.Node`.  The `hasOption` function is comparing the labels as strings. One example of this use is the tables select where label is the name of the table, `customLabel` is a node with an icon and the name, and value is the `id` of the table.

Reply:
I understand the intention but changing it to `ReactNode` will cause too many incompatibility in other references of the `Select` component. Ideally we should probably refactor `customLabel` to `optionalRenderer`. It also doesn't feel right for a data fetching API (the `options` promise) to return `ReactNode`. Data fetching and rendering should be two separate things.